nexus/src/app/silo.rs

```diff
@@ -181,13 +170,10 @@ impl super::Nexus {
     pub async fn silo_user_fetch(
         &self,
         opctx: &OpContext,
-        silo_name: &Name,
+        silo_lookup: &lookup::Silo<'_>,
         silo_user_id: Uuid,
     ) -> LookupResult<db::model::SiloUser> {
-        let (authz_silo,) = LookupPath::new(opctx, self.datastore())
-            .silo_name(silo_name)
-            .lookup_for(authz::Action::Read)
-            .await?;
+        let (authz_silo,) = silo_lookup.lookup_for(authz::Action::Read).await?;
```

Comment:
As discussed in chat with @zephraph, we considered the possibility that this would succeed at lookup up a user by ID as long as the specified silo exists, even if the silo doesn't match the user's ID (because the silo is technically not needed for the lookup if we have a user ID to look up directly). But there is an explicit check that the retrieved user's silo ID matches the one passed in:

https://github.com/oxidecomputer/omicron/blob/7be926a4c4764162f58adaaa16e5565ccecb8b1d/nexus/src/app/silo.rs#L156-L158